client/state/selectors/get-editor-url.js

```diff
@@ -1,26 +1,38 @@
 /**
  * Internal dependencies
  */
-import getGutenbergEditorUrl from 'calypso/state/selectors/get-gutenberg-editor-url';
-import isClassicEditorForced from 'calypso/state/selectors/is-classic-editor-forced';
-import isEligibleForGutenframe from 'calypso/state/gutenberg-iframe-eligible/is-eligible-for-gutenframe';
-import { getSiteAdminUrl } from 'calypso/state/sites/selectors';
+import { isEligibleForGutenframe } from 'calypso/state/gutenberg-iframe-eligible/is-eligible-for-gutenframe';
+import { getSiteAdminUrl, getSiteSlug } from 'calypso/state/sites/selectors';
+import { getEditorPath } from 'calypso/state/editor/selectors';
+import { addQueryArgs } from 'calypso/lib/route';
+import isVipSite from 'calypso/state/selectors/is-vip-site';
 
 export const getEditorUrl = ( state, siteId, postId = null, postType = 'post' ) => {
-	// isEligibleForGutenframe deals with the server side checks for classic editor plugin etc.
-	// isClassicEditerForced has the client side checks. We should combine these!
-	if ( ! isEligibleForGutenframe( state, siteId ) || isClassicEditorForced( state, siteId ) ) {
+	if ( ! isEligibleForGutenframe( state, siteId ) ) {
```

Comment:
The check is whether it's desktop and we need to redirect to wp-admin:

https://github.com/Automattic/wp-calypso/blob/bbad57e43a740e0d5ea37224b9e4114a2ed8beed/client/state/selectors/is-classic-editor-forced.js#L25-L26

So, I was working on the basis that we'll redirect to `wp-admin` based on whether the site is eligible for Gutenframe (and we can ignore whether it's the desktop app or not), because `gutenberg-redirect` and `gutenberg-redirect-and-style` indicate that it's ineligible. I was puzzling over this logic though, so I may be missing something!

Reply:
I think we just need to test it by building the desktop app from this branch. I _think_ there might be a reason why it's set up that way, but I'm not sure.

Reply:
I've compiled and run the desktop app using this branch. Everything works as expected. Simple sites load in the app in the Gutenframe. Eligible Jetpack sites do too, and ineligible ones have the editor open in the browser outside of the app.

There is an issue with VIP. Since the move to VIP Go, the VIP sites are now eligible for Gutenframe. In the browser, we are sent to the usual address, but are shown a message about the migration, and then sent back to the Gutenframe, which loads correctly. The desktop app hangs, and eventually loads the editor in the browser (obviously after some kind of timeout). This happens on `trunk` as well as this branch though, so I think this refactor is fine.